### PR TITLE
ENH: add iso-format support to to_timedelta (#21877)

### LINKED ISSUE #21877: pd.to_timedelta not parsing iso-formatted strings
#### Code Sample, a copy-pastable example if possible

```python
>>> delta = pd.Timedelta(1e9).isoformat()
>>> pd.to_timedelta(delta)
Traceback (most recent call last):
  File "pandas/_libs/tslibs/timedeltas.pyx", line 480, in pandas._libs.tslibs.timedeltas.timedelta_from_spec
    unit = timedelta_abbrevs[unit.lower()]
KeyError: 'p'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sergey/projects/pandas/pandas/core/tools/timedeltas.py", line 99, in to_timedelta
    box=box, errors=errors)
  File "/home/sergey/projects/pandas/pandas/core/tools/timedeltas.py", line 145, in _coerce_scalar_to_timedelta_type
    result = convert_to_timedelta64(r, unit)
  File "pandas/_libs/tslibs/timedeltas.pyx", line 139, in pandas._libs.tslibs.timedeltas.convert_to_timedelta64
    cpdef convert_to_timedelta64(object ts, object unit):
  File "pandas/_libs/tslibs/timedeltas.pyx", line 186, in pandas._libs.tslibs.timedeltas.convert_to_timedelta64
    ts = np.timedelta64(parse_timedelta_string(ts))
  File "pandas/_libs/tslibs/timedeltas.pyx", line 345, in pandas._libs.tslibs.timedeltas.parse_timedelta_string
    r = timedelta_from_spec(number, frac, unit)
  File "pandas/_libs/tslibs/timedeltas.pyx", line 482, in pandas._libs.tslibs.timedeltas.timedelta_from_spec
    raise ValueError("invalid abbreviation: {0}".format(unit))
ValueError: invalid abbreviation: P



```
#### Problem description

#19191 added support for iso-formatted strings to `pd.Timedelta` constructor but `pd.to_timedelta` cannot parse such strings yet.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: bdb61688aa80a6ec9bdb6832946dd83ab4b4c9c8
python: 3.6.6.final.0
python-bits: 64
OS: Linux
OS-release: 4.13.0-45-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.0.dev0+305.gbdb61688a
pytest: 3.6.2
pip: 10.0.1
setuptools: 39.2.0
Cython: 0.28.3
numpy: 1.14.5
scipy: 1.1.0
pyarrow: 0.9.0
xarray: 0.10.7
IPython: 6.4.0
sphinx: 1.7.5
patsy: 0.5.0
dateutil: 2.7.3
pytz: 2018.4
blosc: None
bottleneck: 1.2.1
tables: 3.4.4
numexpr: 2.6.5
feather: 0.4.0
matplotlib: 2.2.2
openpyxl: 2.5.4
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.5
lxml: 4.2.2
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.2.8
pymysql: 0.8.1
psycopg2: None
jinja2: 2.10
s3fs: 0.1.5
fastparquet: 0.1.5
pandas_gbq: None
pandas_datareader: None
gcsfs: 0.1.0

</details>

### PR DESCRIPTION
**Pull request by fjdiod**
- [x] closes #21877
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by WillAyd — CHANGES_REQUESTED** [review]
Can you add an indexing test as well to make sure the function appropriately handles multiple timedeltas?

File: pandas/tests/indexes/timedeltas/test_construction.py [hunk 1]
```diff
@@ -44,6 +44,11 @@ def test_constructor(self):
         tm.assert_index_equal(TimedeltaIndex([400, 450, 1200], unit='ms'),
                               expected)
 
+    def test_constructor_iso(self):
+        expected = timedelta_range('1s', periods=9, freq='s')
+        durations = ['P0DT0H0M{}S'.format(i) for i in range(1, 10)]
+        tm.assert_index_equal(to_timedelta(durations), expected)
```
Comment (WillAyd) [inline pandas/tests/indexes/timedeltas/test_construction.py]: Minor nit but we have a pseudo-standard of doing `tm.assert_index_equal(result, expected)` so would be nice to assign to a variable called `result` in the line above and simply do the comparison on the last line

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/tests/scalar/timedelta/test_timedelta.py [hunk 2]
```diff
@@ -233,6 +233,11 @@ def check(value):
         assert tup.microseconds == 999
         assert tup.nanoseconds == 0
 
+    def test_iso_convertion(self):
```
Comment (WillAyd) [inline pandas/tests/scalar/timedelta/test_timedelta.py]: Minor typo of the word `conversion` if you care to fix up

**Review by WillAyd — APPROVED** [review]

File: pandas/tests/scalar/timedelta/test_timedelta.py (hunk 2 above)
Comment (fjdiod) [reply pandas/tests/scalar/timedelta/test_timedelta.py]: Should I push a fix for it?

Comment (WillAyd) [reply pandas/tests/scalar/timedelta/test_timedelta.py]: Go for it

**Comment by fjdiod** [comment]
Hello, @WillAyd, could I work on #21140 again now?

**Comment by jreback** [comment]
thanks @fjdiod

**Comment by WillAyd** [comment]
Go for it - thanks!

**Pull request merged**